### PR TITLE
IntelliSense Cleanup

### PR DESCRIPTION
**Pull request by bcollamore**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -18,12 +18,36 @@ dotnet_style_qualification_for_event = false:error
...
+dotnet_diagnostic.IDE0052.severity = error
+dotnet_diagnostic.IDE0054.severity = error
+dotnet_diagnostic.IDE0059.severity = error
+dotnet_diagnostic.IDE0060.severity = error
+dotnet_diagnostic.IDE0062.severity = error
+dotnet_diagnostic.IDE0066.severity = error
+dotnet_diagnostic.IDE0074.severity = error
+dotnet_diagnostic.IDE0083.severity = error
+
+# Cannot use index/range with .NET 2.0
```
Comment (ynsehoornenborg) [inline .editorconfig:47]: .NET Standard you mean I guess?

File: Philips.CodeAnalysis.Common/Helper.cs [hunk 2]
```diff
@@ -127,7 +126,7 @@ public static bool HasAttribute(SyntaxList<AttributeListSyntax> attributeLists,
 		public static bool HasAttribute(SyntaxList<AttributeListSyntax> attributeLists, SyntaxNodeAnalysisContext context, string name, string fullName, out Location location, out AttributeArgumentSyntax argumentValue)
 		{
 			location = null;
-			argumentValue = default(AttributeArgumentSyntax);
+			argumentValue = default;
 			if (attributeLists == null)
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.Common/Helper.cs:130]: SyntaxList is a struct, so attributeLists cannot be null

File: Philips.CodeAnalysis.Common/Helper.cs [hunk 3]
```diff
@@ -154,8 +153,7 @@ public static bool HasAttribute(AttributeListSyntax attributes, SyntaxNodeAnalys
 			{
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.Common/Helper.cs:153]: I agree with SonarCloud here, LINQ would help the readability here

File: Philips.CodeAnalysis.Common/Helper.cs [hunk 4]
```diff
@@ -183,12 +181,11 @@ public static bool HasAttribute(AttributeListSyntax attributes, SyntaxNodeAnalys
...
 		{
 			location = null;
-			argument = default(AttributeArgumentSyntax);
+			argument = default;
 
 			if (attribute.Name.ToString().Contains(name))
 			{
-				IMethodSymbol memberSymbol = context.SemanticModel.GetSymbolInfo(attribute).Symbol as IMethodSymbol;
-				if (memberSymbol != null && memberSymbol.ToString().StartsWith(fullName))
+				if (context.SemanticModel.GetSymbolInfo(attribute).Symbol is IMethodSymbol memberSymbol && memberSymbol.ToString().StartsWith(fullName))
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.Common/Helper.cs:188]: Code seems familiar, see Line 156

File: Philips.CodeAnalysis.DuplicateCodeAnalyzer/AvoidDuplicateCodeAnalyzer.cs [hunk 5]
```diff
@@ -674,7 +674,7 @@ public virtual Func<LocationEnvelope> MakeFullLocationEnvelope(TokenInfo firstTo
 
 			(var components, var hash) = _hashCalculator.ToComponentHashes();
 
-			Evidence e = new Evidence(MakeFullLocationEnvelope(firstToken, token), components, hash);
+			Evidence e = new(MakeFullLocationEnvelope(firstToken, token), components, hash);
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.DuplicateCodeAnalyzer/AvoidDuplicateCodeAnalyzer.cs:677]: Not sure if this line got any better to read. Consider a local variable for the result of `MakeFullLocationEnvelope`

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/LogExceptionAnalyzer.cs [hunk 6]
```diff
@@ -90,7 +90,7 @@ public void AnalyzeCatchException(SyntaxNodeAnalysisContext context)
 				var catchNode = (CatchClauseSyntax)context.Node;
 				// Look for logging method calls underneath this node.
 				var hasCallingLogNodes = catchNode.DescendantNodes()
-					.OfType<InvocationExpressionSyntax>().Any(x => IsCallingLogMethod(context, x));
+					.OfType<InvocationExpressionSyntax>().Any(x => IsCallingLogMethod(x));
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/LogExceptionAnalyzer.cs:93]: Consider to have every Linq statement on a separate line for readability

File: Philips.CodeAnalysis.MsTestAnalyzers/AssertMethodCallDiagnosticAnalyzer.cs [hunk 7]
```diff
@@ -30,8 +29,7 @@ private void Analyze(SyntaxNodeAnalysisContext context)
 			{
 				foreach (Diagnostic diagnostic in Analyze(context, invocationExpression, memberAccessExpression) ?? Array.Empty<Diagnostic>())
 				{
-					IMethodSymbol memberSymbol = context.SemanticModel.GetSymbolInfo(memberAccessExpression).Symbol as IMethodSymbol;
-					if ((memberSymbol == null) || !memberSymbol.ToString().StartsWith("Microsoft.VisualStudio.TestTools.UnitTesting.Assert"))
+					if ((context.SemanticModel.GetSymbolInfo(memberAccessExpression).Symbol is not IMethodSymbol memberSymbol) || !memberSymbol.ToString().StartsWith("Microsoft.VisualStudio.TestTools.UnitTesting.Assert"))
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.MsTestAnalyzers/AssertMethodCallDiagnosticAnalyzer.cs:32]: OK, seen this code snippet too often now. Consider putting this code into a Utility / Helper

File: Philips.CodeAnalysis.MsTestAnalyzers/TestContextAnalyzer.cs [hunk 8]
```diff
@@ -79,14 +77,14 @@ private static void Analyze(SyntaxNodeAnalysisContext context)
...
-					!(context.SemanticModel.GetSymbolInfo(identifier).Symbol is ITypeSymbol))
+					context.SemanticModel.GetSymbolInfo(identifier).Symbol is not ITypeSymbol)
 				{
 					// if we find the same identifier as the propery and it's not a type or the original instance, it's used
 					return;
 				}
 
-				if ((identifier.Identifier.ToString() == varName) && !(identifier.Parent is VariableDeclarationSyntax) &&
-					!propNodes.Contains(identifier) && !(context.SemanticModel.GetSymbolInfo(identifier).Symbol is ITypeSymbol))
+				if ((identifier.Identifier.ToString() == varName) && identifier.Parent is not VariableDeclarationSyntax &&
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.MsTestAnalyzers/TestContextAnalyzer.cs:86]: Consider putting each condition on its own line for readability

File: Philips.CodeAnalysis.Test/Helpers/CodeFixVerifier.Helper.cs [hunk 9]
```diff
@@ -79,7 +79,7 @@ private static string GetStringFromDocument(Document document)
 			SyntaxNode syntaxNode = document.GetSyntaxRootAsync().Result;
 			if (syntaxNode != null)
 			{
-				var vv = syntaxNode.GetText().ToString();
+				_ = syntaxNode.GetText().ToString();
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.Test/Helpers/CodeFixVerifier.Helper.cs:82]: Seems like a bug to me that this result is not used

**Comment by ynsehoornenborg** [comment]
Nice update, good to see some modern C# syntax being used

**Review by ynsehoornenborg — DISMISSED** [review]

File: Philips.CodeAnalysis.Common/Helper.cs (hunk 4 above)
Comment (bcollamore) [reply Philips.CodeAnalysis.Common/Helper.cs:188]: Agreed, but I don't want to touch the logic in this large PR.

File: Philips.CodeAnalysis.DuplicateCodeAnalyzer/AvoidDuplicateCodeAnalyzer.cs (hunk 5 above)
Comment (bcollamore) [reply Philips.CodeAnalysis.DuplicateCodeAnalyzer/AvoidDuplicateCodeAnalyzer.cs:677]: Fixed, and created Issue to investigate automation.

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/LogExceptionAnalyzer.cs (hunk 6 above)
Comment (bcollamore) [reply Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/LogExceptionAnalyzer.cs:93]: Fixed, and filed Issue to explore automation

File: Philips.CodeAnalysis.MsTestAnalyzers/TestContextAnalyzer.cs (hunk 8 above)
Comment (bcollamore) [reply Philips.CodeAnalysis.MsTestAnalyzers/TestContextAnalyzer.cs:86]: Fixed, and filed Issue for automation

File: Philips.CodeAnalysis.Test/Helpers/CodeFixVerifier.Helper.cs (hunk 9 above)
Comment (bcollamore) [reply Philips.CodeAnalysis.Test/Helpers/CodeFixVerifier.Helper.cs]: I'm guessing it was there for debugging purposes only. Removed the code. Add Issue to investigate automation.

File: Philips.CodeAnalysis.MsTestAnalyzers/AssertMethodCallDiagnosticAnalyzer.cs (hunk 7 above)
Comment (bcollamore) [reply Philips.CodeAnalysis.MsTestAnalyzers/AssertMethodCallDiagnosticAnalyzer.cs:32]: Beyond scope of this large already commit.

**Review by ynsehoornenborg — APPROVED** [review]

**Pull request merged**